### PR TITLE
Update markdown docs formatting

### PR DESCRIPTION
**Pull request by calebmshafer**
Update VSCode settings to have markdownlint fix formatting on save.  Also batch ran it on all markdown files in the docs folder.

One of the main changes, other than running markdownlint, is to use `-` instead of `*` for all unordered lists. We had a mix  of them even within the same files.  Unfortunately markdownlint doesn't provide a rule for that, or an automatic fix, so we'll need to find something else to make it consistent moving forward.

### CODE REVIEW

**Review by scsewall — APPROVED** [review]

**Comment by calebmshafer** [comment]
@williamkbentley mind taking a look at the changes?

File: docs/learning/backend/SchemasAndElementsInTypeScript.md [hunk 1]
```diff
@@ -28,7 +28,7 @@ The app can ensure that the underlying schema is imported by registering an onOp
 
 where the schema is:
 
-``` xml
+```xml
```
Comment (williamkbentley) [inline docs/learning/backend/SchemasAndElementsInTypeScript.md:31]: Why does this one have to be formatted this way, but the others on the page are okay?

**Review by williamkbentley — CHANGES_REQUESTED** [review]

**Review by williamkbentley — APPROVED** [review]
Looks good

File: docs/learning/backend/SchemasAndElementsInTypeScript.md (hunk 1 above)
Comment (calebmshafer) [reply docs/learning/backend/SchemasAndElementsInTypeScript.md]: doesn't have to be, I think I manually switched that one around.

**Pull request merged**